arcgis-ios-sdk-samples/Edit data/Edit with branch versioning/EditWithBranchVersioningViewController.swift

```diff
@@ -0,0 +1,524 @@
+        alertController.popoverPresentationController?.sourceRect = sourceRect
+        present(alertController, animated: true)
+    }
+    
+    func chooseVersion(_ sender: UIBarButtonItem, completion: @escaping (String) -> Void) {
+        let alertController = UIAlertController(
+            title: "Choose to switch to another version",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        existingVersionNames.forEach { name in
+            let action = UIAlertAction(title: name, style: .default) { _ in
+                completion(name)
+            }
+            alertController.addAction(action)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    func chooseVersionAccessPermission(_ sender: UIBarButtonItem, completion: @escaping (AGSVersionAccess) -> Void) {
+        let alertController = UIAlertController(
+            title: "Choose an access level for the new branch version",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        let versionAccessPermission: KeyValuePairs<String, AGSVersionAccess> = [
+            "Public": .public,
+            "Protected": .protected,
+            "Private": .private
+        ]
+        versionAccessPermission.forEach { name, permission in
+            let action = UIAlertAction(title: name, style: .default) { _ in
+                completion(permission)
+            }
+            alertController.addAction(action)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames  = ["EditWithBranchVersioningViewController"]
+        // Load the service geodatabase.
+        let damageFeatureService = URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer")!
+        serviceGeodatabase = loadServiceGeodatabase(from: damageFeatureService)
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+
+extension EditWithBranchVersioningViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Disable interaction with features on default branch.
+        if currentVersionName == defaultVersionName {
+            return
+        }
+        // Dismiss any presenting callout.
+        mapView.callout.dismiss()
```

Comment:
Is there a reason why you disabled interaction with features on the default branch? I think it would be helpful if the user could select a feature and view its name and status in order to compare it with the edited branches.

Reply:
I remember Jared mentioned he disabled interaction on default branch last Friday, so I followed in that way. But after some experiments, I found your suggestion more reasonable, so I made some updates 👇 .